### DISCUSSION TITLE
Not all messages are showing up for JSONSchema messages

### DISCUSSION THREAD

**Discussion by johhue** [tchiotludo/akhq#1311] [Q&A] [answered]
Hello I have a topic whereby on the TOPIC LIST page in AKHQ it says there are 8 messages.

However, when I click on the topic to show all the detail, not all messages are shown and sometimes no messages are shown.

I can see from multiple places that there are 8 messages. I can see on the metrics page there are 8 messages.

kafka_log_log_value{instance="myserver:1111",job="kafka",name="LogEndOffset",partition="0",topic="my.topic"} 219
kafka_log_log_value{instance="myserver:1111",job="kafka",name="LogStartOffset",partition="0",topic="my.topic"} 211

This only seems to happen for messages which are serialized with JSONSchema.

Is there any way to add debugging to show why this is happening? Is it tripping up on some control character that the JSONSchema serialization is doing?

So my questions:

How does AKHQ arrive at the value of 8. I see the "around" symbol there instead of an equal symbol.
How often is this number updated? Is it live or cached periodically?

Why is when I show the detail there are not 8 messages.

Thank you,
John

**Comment by tchiotludo** [comment, marked as answer]
Is the topic compacted ? if yes, it's normal, the counter is an approximation based on begin and end offset, if the topic are compacted, the number is not relevant and can't be relevant, since there is no Kafka Api that return the exact value.
